docs/testspec/src/12_pubkey_kem.rst

```diff
@@ -1,6 +1,124 @@
+   |                        |    resulting shared secret and ciphertext to expected values in the     |
+   |                        |    test vector. Again, the ciphertext is hashed to save disk space.     |
+   |                        |                                                                         |
+   |                        | #. Decapsulate the just-calculated ciphertext with the private key from |
+   |                        |    the encode/decode roundtrip and ensure that the resulting shared     |
+   |                        |    secret is equal to the expected value from the test vector.          |
+   +------------------------+-------------------------------------------------------------------------+
+
+.. table::
+   :class: longtable
+   :widths: 20 80
+
+   +------------------------+-------------------------------------------------------------------------+
+   | **Test Case No.:**     | PKENC-CMCE-2                                                            |
+   +========================+=========================================================================+
+   | **Type:**              | Negative Test/Known Answer Tests                                        |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Description:**       | For a plaintext confirmation (pc) and a non-pc instance:                |
+   |                        | Generate a Classic McEliece keypair using a KAT seed, use the private   |
+   |                        | key to decapsulate an invalid ciphertext, and compare the resulting     |
+   |                        | value with the value generated using the reference implementation.      |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Preconditions:**     | None                                                                    |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Input Values:**      | KAT seed, invalid ciphertext, corresponding invalid shared secret       |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Expected Output:**   | The correct invalid shared secret                                       |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Steps:**             | For each available algorithm parameterization:                          |
+   |                        |                                                                         |
+   |                        | #. Generate a key pair using the KAT seed with AES-256-CTR-DRBG         |
+   |                        |                                                                         |
+   |                        | #. Decapsulate the invalid ciphertext (input value).                    |
```

Comment:
How do we get invalid ciphertexts? Are they provided by the KAT vector? If so, how did we make it invalid?

Reply:
> How do we get invalid ciphertexts?

They were generated using fabian.exe ;) I used the valid ciphertext and flipped a single bit. I did this once with the first bit and once with the last bit (to cover a case where only $C_1$ of plaintext confirmation is wrong). I described it briefly in the text above. But you are right, I'll describe it here.